Remove custom astral

scliddle opened the pull request:
removed the custom astral.
cleaned up beginning of file
added docstring and version
added client-side HoP LocationInfo
added helper function
verified output is within a few seconds 

[after_sunsets_2020.txt](https://github.com/basil96/civilite/files/4500459/after_sunsets_2020.txt)
[before_sunsets_2020.txt](https://github.com/basil96/civilite/files/4500460/before_sunsets_2020.txt)

scliddle commented:
converted to an observer to gain elevation, but now the differences are closer to a minute from before. please review @basil96 
[sunsets_2020.txt](https://github.com/basil96/civilite/files/4500547/sunsets_2020.txt)

basil96 commented on src/schedule.py:
```diff
@@ -39,17 +39,17 @@ def AddEvent(self, weekday, event):
 
 
-def get_civil_twilight(city, event_date):
+def get_civil_twilight(observer, event_date, observer_timezone):
```
Stay consistent with namingStyle.

basil96 replied:
I've gotten used to camelCase for function names, and snake_case for args.  I don't know what's better or if anyone cares.

scliddle replied:
okay. I'll fix it

basil96 commented on src/schedule.py:
```diff
@@ -58,20 +58,20 @@ def getEventType(schedule, event_date, city):
-    # city = a['Rochester']     # custom addition: Rochester,USA,43°10'N,77°36'W,US/Eastern,153
-    city = LocationInfo('Rochester_HoP', region='USA', timezone='US/Eastern', latitude=43.1606355, longitude=-77.3883843) # custom addition: Rochester_HoP,USA,43°09'N,77°23'W,US/Eastern,170
+    '''create events from a schedule object at a custom location'''
+    # custom observer: Rochester_HoP,USA,43°09'N,77°23'W,US/Eastern,170
+    observer_location = Observer(latitude=43.1606355, longitude=-77.3883843, elevation=170)
+    tz = timezone('US/Eastern')
```
This is good. Eventually we can roll these functions into a class, allowing caller to feed custom observer and tz.

basil96 replied:
Also, **observer_location** can just be named **observer**.

scliddle replied:
because the Class name was observer, I opted for a more verbose name. I am okay with observer.

scliddle replied:
I think astral.Observer should have a timezone member. Maybe we can add an issue to astral

basil96 replied:
> I think astral.Observer should have a timezone member. Maybe we can add an issue to astral

I had a similar thought, but looking over their **__main__.py** example and **sun.py**, I don't think they'd go for that because it looks like **LocationInfo** is intended to take a TZ as an arg, while **Observer** is a representation of a time-agnostic location.  All the functions in their **sun** module take an Observer, a date, and a tzinfo already.  The change would be API-breaking.

basil96 commented:
> converted to an observer to gain elevation, but now the differences are closer to a minute from before. please review @basil96
> [sunsets_2020.txt](https://github.com/basil96/civilite/files/4500547/sunsets_2020.txt)

I missed this because I think it came in while I was in the middle of reviewing the previous commit.
I think the couple minutes of shift is okay.  Sunset event extended by one Tuesday in October, and one Sunday in November reverted back to Sunset relative to the "before" change.  These are minute minute changes.  The event types only exist because they affect the weeks when I have to manually reprogram the Intermatic.  We're trying to obsolete that unit, so the event type will eventually become moot.  Whatever the on/off times work out to be based on astral's logic, that will be the rule.
From this point on we'll need a unit test for this that would verify a year's worth of events within a tolerance.  Writing to fpOut can be optional, I've only been using that as a diagnostic.

basil96 commented on src/schedule.py:
```diff
@@ -38,18 +38,23 @@ def AddEvent(self, weekday, event):
+class ScheduleObserver(Observer):
+    '''schedule implementation of astral.Observer to aggregate timezone'''
+    timezone: str = "UTC"
+    def __init__(self, timezone="UTC", **kwargs):
```
General use is to pass in a tzinfo object like so: `tzinfo: datetime.tzinfo = pytz.utc` instead of strings.

scliddle replied:
okay, see my most recent push. I like this better. I chose pytz over datetime as astral uses pytz timezones.

scliddle commented:
I think I pulled off what we were going for. I added a class **ScheduleObserver** that has a timezone attribute and also a twilight member function. fixed the naming issues you mentioned. @basil96

basil96 commented on src/schedule.py:
```diff
@@ -38,18 +38,23 @@ def AddEvent(self, weekday, event):
-    return twilight_start_end[0]
-
-def getEventType(schedule, event_date, observer, observer_timezone):
+class ScheduleObserver(Observer):
```
I like this.

basil96 commented:
Trying a squash merge of #2.
